src/main/java/jenkinsci/plugins/influxdb/generators/SonarQubePointGenerator.java

```diff
@@ -183,19 +200,19 @@ String getSonarProjectName(String url) throws URISyntaxException {
         return projectUrl.length > 1 ? projectUrl[projectUrl.length - 1] : "";
     }
 
-    private int getLinesOfCode(String url) throws IOException {
+    private float getSonarMetric(String url, String metric) throws IOException {
         String output = getResult(url);
         JSONObject metricsObjects = JSONObject.fromObject(output);
-        int linesOfCodeCount = 0;
+        float value = 0;
         JSONArray array = metricsObjects.getJSONObject("component").getJSONArray("measures");
         for (int i = 0; i < array.size(); i++) {
```

Comment:
Shouldn't there be an early exit in the loop when the matching metric was found?